ql/src/java/org/apache/hadoop/hive/ql/io/thrift/twitter/ThriftGeneralInputFormat.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hive.ql.io.thrift.twitter;
+
+import com.twitter.elephantbird.mapred.input.DeprecatedFileInputFormatWrapper;
+import com.twitter.elephantbird.mapreduce.input.MultiInputFormat;
+import com.twitter.elephantbird.mapreduce.io.BinaryWritable;
+import com.twitter.elephantbird.util.TypeRef;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.ql.io.CombineHiveInputFormat;
+import org.apache.hadoop.hive.serde.serdeConstants;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.mapred.FileSplit;
+import org.apache.hadoop.mapred.InputSplit;
+import org.apache.hadoop.mapred.JobConf;
+import org.apache.hadoop.mapred.RecordReader;
+import org.apache.hadoop.mapred.Reporter;
+
+import java.io.IOException;
+
+@SuppressWarnings("deprecation")
+public class ThriftGeneralInputFormat
+    extends DeprecatedFileInputFormatWrapper<LongWritable, BinaryWritable>
+    implements CombineHiveInputFormat.AvoidSplitCombination
+{
+  public ThriftGeneralInputFormat()
+  {
+    super(new MultiInputFormat());
+  }
+
+  private void initialize(FileSplit split, JobConf job) throws IOException
+  {
+    String thriftClassName = job.get(serdeConstants.SERIALIZATION_CLASS);
```

Comment:
We do not need to. I was just wondering if later people start using a different TBase Class in Table definition. At that point using ThriftGenericRow might not be correct. It might still work and fail somewhere else. Do you think that is a valid concern?